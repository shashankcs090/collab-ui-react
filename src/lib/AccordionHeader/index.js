/**
 * @category layout
 * @component accordion
 * @variations collab-ui-react
 */

import React from 'react';
import PropTypes from 'prop-types';

export default class AccordionHeader extends React.Component {
  static displayName = 'AccordionHeader';

  static contextTypes = {
    onClick: PropTypes.func,
    onKeyDown: PropTypes.func
  };

  componentDidUpdate(prevProps) {
    const { focus } = this.props;
    !prevProps.focus && focus && this.headerRef.focus();
  }

  render() {
    const {
      children,
      className,
      disabled,
      showSeparator,
    } = this.props;

    const { onClick, onKeyDown } = this.context;

    return (
      <div
        role='button'
        ref={ref => this.headerRef = ref}
        onClick={onClick}
        onKeyDown={onKeyDown}
        tabIndex={!disabled ? 0 : -1}
        className={
          'cui-accordion__header' +
          `${(showSeparator && ` cui-accordion__header--separator`) || ''}` +
          `${(className && ` ${className}`) || ''}`
        }
      >
          {children}
          <span className="cui-accordion__header-icon"/>
      </div>
    );
  }
}

AccordionHeader.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  focus: PropTypes.bool,
  showSeparator: PropTypes.bool,
};

AccordionHeader.defaultProps = {
  children: null,
  className: '',
  disabled: false,
  focus: false,
  showSeparator: false,
};
