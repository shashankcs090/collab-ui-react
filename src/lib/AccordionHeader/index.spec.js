import React from 'react';
import { shallow, mount } from 'enzyme';
import AccordionHeader from '../AccordionHeader';

describe('tests for <AccordionHeader />', () => {
  it('should match SnapShot', () => {
    const container = shallow(<AccordionHeader />);

    expect(container).toMatchSnapshot();
  });

  it('should render children', () => {
    const container = shallow(
      <AccordionHeader>
        <div className="testingforTC" />
      </AccordionHeader>
    );
    expect(container.find('.testingforTC').length).toEqual(1);
  });

  it('should pass ClassName Prop', () => {
    const container = shallow(<AccordionHeader className="testClass"/>);

    expect(container.find('.testClass').exists()).toEqual(true);
    expect(container.find('.cui-accordion__header').props().tabIndex).toEqual(0);
  });

  it('when the header is disabled, should set tabIndex to -1', () => {
    const container = shallow(<AccordionHeader disabled />);
    expect(container.find('.cui-accordion__header').props().tabIndex).toEqual(-1);
  });

  it('when the showSeparator prop is true, should display the separator', () => {
    const container = shallow(<AccordionHeader disabled showSeparator/>);
    expect(container.find('.cui-accordion__header').hasClass('cui-accordion__header--separator')).toEqual(true);
  });

  it('should call the context handler function on click and keyDown event', () => {
    const onClickFn = jest.fn();
    const onKeyDownFn = jest.fn();
    const container = mount(<AccordionHeader/>, {
      context: {
        onClick: onClickFn,
        onKeyDown: onKeyDownFn,
      }
    });
    container.simulate('click');
    expect(onClickFn).toHaveBeenCalled();
    container.simulate('keyDown');
    expect(onKeyDownFn).toHaveBeenCalled();
  });

});
