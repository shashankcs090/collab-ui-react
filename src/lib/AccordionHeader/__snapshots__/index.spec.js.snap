// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <AccordionHeader /> should match SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AccordionHeader
    className=""
    disabled={false}
    focus={false}
    showSeparator={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <span
          className="cui-accordion__header-icon"
        />,
      ],
      "className": "cui-accordion__header cui-accordion__header--separator",
      "onClick": undefined,
      "onKeyDown": undefined,
      "role": "button",
      "tabIndex": 0,
    },
    "ref": [Function],
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "cui-accordion__header-icon",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <span
            className="cui-accordion__header-icon"
          />,
        ],
        "className": "cui-accordion__header cui-accordion__header--separator",
        "onClick": undefined,
        "onKeyDown": undefined,
        "role": "button",
        "tabIndex": 0,
      },
      "ref": [Function],
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "cui-accordion__header-icon",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
